### PR TITLE
Add support for removing config entry from a device

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Add support for removing config entry from a device

Depends on core PR: https://github.com/home-assistant/core/pull/66188

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/devices/ha-config-device-page.ts [hunk 1]
```diff
@@ -263,6 +269,65 @@ export class HaConfigDevicePage extends LitElement {
...
+    });
+
+    buttons = buttons.filter(Boolean);
+
+    if (buttons.length > 0) {
+      this._deleteButtons = (
+        buttons as { entry_id: string; domain: string }[]
+      ).map(
+        (button) => html`
+          <mwc-button class="warning" @click=${this._confirmDeleteEntry}>
```
Comment (emontnemery) [inline src/panels/config/devices/ha-config-device-page.ts:299]: We should pass in the specific config entry id from `button.entry_id`, how to do that?

File: src/panels/config/devices/ha-config-device-page.ts [hunk 2]
```diff
@@ -198,7 +202,9 @@ export class HaConfigDevicePage extends LitElement {
     }
 
     this._diagnosticDownloadLinks = Math.random();
+    this._deleteButtons = [".."];
```
Comment (emontnemery) [inline src/panels/config/devices/ha-config-device-page.ts:205]: Added as a hack to not keep re-rendering if there are no delete buttons. What would be the correct way?

Comment (balloob) [reply src/panels/config/devices/ha-config-device-page.ts:205]: An empty list is truthy.

File: src/panels/config/devices/ha-config-device-page.ts [hunk 3]
```diff
@@ -263,6 +269,66 @@ export class HaConfigDevicePage extends LitElement {
...
+                      this.hass.localize,
+                      button.domain
+                    ),
+                  }
+                )
+              : this.hass.localize(`ui.panel.config.devices.delete_device`)}
+          </mwc-button>
+        `
+      );
+    }
```
Comment (bramkragten) [inline src/panels/config/devices/ha-config-device-page.ts:315]:
```suggestion
      this._deleteButtons = buttons;
    }
```

File: src/panels/config/devices/ha-config-device-page.ts [hunk 4]
```diff
@@ -263,6 +269,66 @@ export class HaConfigDevicePage extends LitElement {
...
+    buttons = buttons.filter(Boolean);
+
+    if (buttons.length > 0) {
+      this._deleteButtons = (
+        buttons as { entry_id: string; domain: string }[]
+      ).map(
+        (button) => html`
+          <mwc-button
+            class="warning"
+            entry_id=${button.entry_id}
```
Comment (bramkragten) [inline src/panels/config/devices/ha-config-device-page.ts:298]:
```suggestion
            .entryId=${button.entry_id}
```

File: src/panels/config/devices/ha-config-device-page.ts [hunk 5]
```diff
@@ -263,6 +269,66 @@ export class HaConfigDevicePage extends LitElement {
...
+                )
+              : this.hass.localize(`ui.panel.config.devices.delete_device`)}
+          </mwc-button>
+        `
+      );
+    }
+  }
+
+  private async _confirmDeleteEntry(e: MouseEvent): Promise<void> {
+    const entry_id = (e.currentTarget! as HTMLElement).getAttribute("entry_id");
```
Comment (bramkragten) [inline src/panels/config/devices/ha-config-device-page.ts:319]:
```suggestion
    const entry_id = (e.currentTarget as any). entryId;
```

File: src/panels/config/devices/ha-config-device-page.ts [hunk 6]
```diff
@@ -263,6 +269,66 @@ export class HaConfigDevicePage extends LitElement {
...
+
+    const confirmed = await showConfirmationDialog(this, {
+      text: this.hass.localize("ui.panel.config.devices.confirm_delete"),
+    });
+
+    if (!confirmed) {
+      return;
+    }
+
+    await removeConfigEntryFromDevice(this.hass!, this.deviceId, entry_id!);
```
Comment (bramkragten) [inline src/panels/config/devices/ha-config-device-page.ts:329]:
```suggestion
    await removeConfigEntryFromDevice(this.hass!, this.deviceId, entryId!);
```

File: src/panels/config/devices/ha-config-device-page.ts [hunk 7]
```diff
@@ -375,6 +432,9 @@ export class HaConfigDevicePage extends LitElement {
     if (Array.isArray(this._diagnosticDownloadLinks)) {
       deviceActions.push(...this._diagnosticDownloadLinks);
     }
+    if (Array.isArray(this._deleteButtons)) {
```
Comment (bramkragten) [inline src/panels/config/devices/ha-config-device-page.ts:435]: It will always be an array, or undefined, so this check can be simpler

Comment (emontnemery) [reply src/panels/config/devices/ha-config-device-page.ts:433]: Makes sense, replaced with:
```diff
-    if (Array.isArray(this._deleteButtons)) {
+    if (this._deleteButtons) {
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**